app/src/main/java/org/simple/clinic/ReplayUntilScreenIsDestroyed.kt

```diff
@@ -0,0 +1,45 @@
+package org.simple.clinic
+
+import io.reactivex.Observable
+import io.reactivex.ObservableSource
+import io.reactivex.ObservableTransformer
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.rxkotlin.ofType
+import io.reactivex.rxkotlin.plusAssign
+import org.simple.clinic.widgets.ScreenDestroyed
+import org.simple.clinic.widgets.UiEvent
+
+class ReplayUntilScreenIsDestroyed(
+    private val transforms: List<ObservableTransformer<UiEvent, UiEvent>> = emptyList()
+) : ObservableTransformer<UiEvent, UiEvent> {
+
+  constructor(transform: ObservableTransformer<UiEvent, UiEvent>) : this(transforms = listOf(transform))
```

Comment:
Let's do `@Suppress("unused")`?

Reply:
I could do it, but we'll end up removing it almost immediately because we can migrate our other screens to it (and `ReportAnalyticsEvents()` will always be present).

Do you still feel we should do it?

Reply:
Discussed offline: This will no longer be relevant when the new API is done.